cmd/snap-confine/mount-support.c

```diff
@@ -581,7 +581,8 @@ void sc_populate_mount_ns(struct sc_apparmor *apparmor, int snap_update_ns_fd,
 					die("cannot locate the core or legacy core snap (current symlink missing?)");
 				}
 			}
-			die("cannot locate the base snap: %s", base_snap_name);
+			if (access(rootfs_dir, F_OK) != 0)
```

Comment:
Can you explain this? I don't see how this is required.

Reply:
Sure, the code above may change rootfs_dir (the fallback code that switches to ubuntu-core) so we need to re-check here if things are still inaccessible .

Reply:
Yes, what @zyga said. Note that your added conditional will always be true. Ie, with this change you end up with:
```
if (access(rootfs_dir, F_OK) != 0) {
   if (sc_streq(base_snap_name, "core")) {
        ... updates rootfs_dir ...
        if (access(rootfs_dir, F_OK) != 0) {
            die
        }
    }
    if (access(rootfs_dir, F_OK) != 0) { # your addition is always true
        die
    }
}
```

Reply:
Ah, I see this too now. Thank you!